File: src/main/java/murphy/software/games/platform2d/move/movement/MovementController.java
package murphy.software.games.platform2d.move.movement;

import murphy.software.games.platform2d.move.model.Actor2D;
import murphy.software.games.platform2d.move.model.Position2D;

public class MovementController {

    public void move(Actor2D actor2D) {
        Position2D xy = DistanceController.distance(actor2D.feetPerSec(), actor2D.angle2D().angle());
        actor2D.changeX(xy.x());
        actor2D.changeY( - xy.y());
    }
}

File: src/main/java/murphy/software/games/platform2d/move/movement/MovementController.kt
package murphy.software.games.platform2d.move.movement

import murphy.software.games.platform2d.move.model.Actor2D

class MovementController {

    companion object {
        val distanceController : DistanceController = DistanceController()
    }
    var companion = Companion
    fun move(actor2D: Actor2D) {
        val xy = distanceController.distance(actor2D.feetPerSec, actor2D.angle2D.angle)
        actor2D.changeX(xy.x)
        actor2D.changeY(-xy.y)
    }
}